core/synch.c

```diff
@@ -1830,11 +1829,13 @@ send_all_other_threads_native(void)
 #endif
 
     /* Suspend all threads except those trying to synch with us */
-    DEBUG_DECLARE(ok =)
-        synch_with_all_threads(desired_state, &threads, &num_threads,
-                               THREAD_SYNCH_NO_LOCKS_NO_XFER,
-                               THREAD_SYNCH_SUSPEND_FAILURE_IGNORE);
-    ASSERT(ok);
+    if (!synch_with_all_threads(desired_state, &threads, &num_threads,
+                                THREAD_SYNCH_NO_LOCKS_NO_XFER,
+                                THREAD_SYNCH_SUSPEND_FAILURE_IGNORE)) {
+        fatal_error(COULDNT_SYNCH_WITH_ALL_THREADS, 1,
+                    "send_all_other_threads_native");
```

Comment:
ASSERT_NOT_REACHED

Reply:
Renamed fatal_error to REPORT_FATAL_ERROR_AND_EXIT and put the assert there.